src/Compilers/CSharp/Portable/Binder/Lifetime.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// A representation of the program region in which a 'ref' can be used,
+    /// which is limited to what is expressible in C#.
+    /// </summary>
+    /// <remarks>
+    /// For example, in this design, all lifetimes have a known relationship to all other lifetimes.
+    /// </remarks>
+    internal readonly struct Lifetime
```

Comment:
>Lifetime [](http://example.com/codeflow?start=29&length=8)

nit: consider `RefLifetime` to narrow down the concept a bit

Reply:
I am hesitant to use Ref in the type name at this time, because there is a temptation to continue using the naming scheme in the spec of *lifetime* versus *ref-lifetime* of an expression (which currently are called *safe-context* and *ref-safe-context*).

I am adjusting the doc comment to try to improve clarity.